File: 14. Longest Common Prefix/longest_common_prefix.py
def longest_common_prefix (array):
	out = ""
	shortest_word = 0
	for a in range(len(array)):
		if (a == 0):
			shortest_word = len (array[a])
		
		if ((0 - len(array[a])) > (-1 * shortest_word)):
			shortest_word = len(array[a])
	
	
	for e in range(shortest_word):
		letter = ""
		common = True
		for a in range(len(array)):
			if (a == 0):
				letter = array[a][e]
			else:
				if letter != array[a][e]:
					#make sure it is not appended
					common = False
					break
				
		if common:
			out += letter
			common = True
			
	return out
	
strs = ["flower","flow","flight"]
print(longest_common_prefix(strs))

File: 14. Longest Common Prefix/longest_common_prefix.php
<?php
function longestCommonPrefix($list){
	$out = "";
	$shortestWordLen = 0;
	
	foreach ($list as $i){
		if ($i == $list[0]){
			$shortestWordLen = strlen($i);
		}
		
		if ((0 - strlen($i)) > (-1 * $shortestWordLen)){
			$shortestWordLen = strlen($i);
		}
	}
	
	for ($letterIndex = 0; $letterIndex < $shortestWordLen; $letterIndex++){
		$letter = "";
		$common = true;
		
		foreach ($list as $word){
			if ($word == $list[0]){
				$letter = $word[$letterIndex];
			} else {
				if ($letter != $word[$letterIndex]){
					$common = false;
					break;
				}
			}
		}
		
		if ($common){
			$out .= $letter;
			$common = true;
		}
	}
	
	return ($out);
}

$string = ["flower", "flow", "flight"];
echo (longestCommonPrefix($string));
echo ("\n\n\n\n");
?>
